membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -41,8 +44,10 @@ class AccountController extends LazyLogging {
       user <- EitherT(Future.successful( maybeUserId \/> "no identity cookie for user"))
       sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
       subscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      account <- EitherT(tp.zuoraService.getAccount(subscription.accountId).map(x => Some(x)).map(_ \/> s"no account for subscription: ${subscription.name} with account id ${subscription.accountId}"))
+      stripeService = if (account.paymentGateway.contains(tp.auStripeService.paymentGateway.gatewayName)) tp.auStripeService else tp.ukStripeService
```

Comment:
should we make the reader do this? I.e. turn the String into a sealed trait ADT to represent which.  Then we can pattern match on that.  Just a thought.

Reply:
OK let's have a chat on Monday on how to do that. I did think about an implicit class over Account.